### PR TITLE
Database implementation

### PR DESCRIPTION
**Pull request by giopera**
Implementation of the code for the management of the database and the fetch of the quotes from it

### CODE REVIEW

**Comment by seajee** [comment]
Implementation status:
- Connecting to the database fails
- Environment variables are not well established

**Comment by seajee** [comment]
Current implementation status:
- Fetching latest quote from client -> server -> database works
- Fetching a quote by index out of bounds crashes the server

**Comment by seajee** [comment]
Every edge case is now handled. Ready for review

File: README.md [hunk 1]
```diff
@@ -4,6 +4,13 @@ A website for showing quotes, everyday.
...
 ## Build and run
 
+This project uses MariaDB as it's database. To setup a container for it run the following
+
+```bash
+mkdir db
+docker-compose up -d
+```
+
 To build and run the project run the following
```
Comment (seajee) [inline README.md:15]: Should add instructions for dotenv file creation

File: src/api.ts [hunk 2]
```diff
@@ -10,32 +10,30 @@ apiRouter.use((_req, res, next) => {
...
+        res.send({
+            quote: quote
         });
-        return;
-    }
-
-    res.send({
-        quote: quotes[index]
     });
+
```
Comment (seajee) [inline src/api.ts:35]: Blank line break

**Comment by seajee** [comment]
LGTM!

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  db:
+    image: "docker.io/mariadb:${MARIADB_VERSION}"
+    restart: always
+    ports:
+      - "${DB_PORT}:3306"
+    volumes:
+      - ./db:/var/lib/mysql 
+    environment:
+      MARIADB_USER: "${DB_USER}"
+      MARIADB_ROOT_PASSWORD: "${DB_ROOT_PASSWORD}"
```
Comment (giopera) [inline docker-compose.yml:13]: Not the intended use for environment variables but it's ok anyway

Comment (seajee) [reply docker-compose.yml:13]: It actually is!
Checkout this docs: https://docs.docker.com/compose/environment-variables/set-environment-variables/

File: src/api.ts [hunk 4]
```diff
@@ -10,32 +10,30 @@ apiRouter.use((_req, res, next) => {
...
         res.status(400).send({
             error: "Index is NaN"
         });
         return;
     }
 
-    if (index < 0 || index >= quotes.length) {
-        res.status(406).send({
-            error: "Out of Bounds"
+    fetchQuote(id, (quote) => {
```
Comment (giopera) [inline src/api.ts:30]: What is the behavior if the id is not in the db?
Should we make it so that a error response is sent in this case?

**Review by giopera — COMMENTED** [review]
Finished review, we need to decide for that response in case of error and maybe other responses in the code

File: docker-compose.yml (hunk 3 above)
Comment (giopera) [reply docker-compose.yml:13]: i'm talking about the "${DB_USER}" and "${DB_ROOT_PASSWORD}" part

File: src/api.ts (hunk 4 above)
Comment (seajee) [reply src/api.ts:30]: You are right. Atm errors are handled by just returning an alternative string as the quote. Error handling should be improved in a future branch. Closing

File: docker-compose.yml (hunk 3 above)
Comment (seajee) [reply docker-compose.yml:13]: Closing as it is intended.

**Comment by seajee** [comment]
Yes, we're going to create a branch for better error handling. Merging.

File: docker-compose.yml (hunk 3 above)
Comment (giopera) [reply docker-compose.yml:13]: Fuck you

**Comment by giopera** [comment]
Huston, db branch merged

**Pull request merged**